api/core/v1/envoy.proto

```diff
@@ -0,0 +1,10 @@
+syntax = "proto3";
+
+package clutch.core.v1;
+
+option go_package = "github.com/lyft/clutch/backend/api/core/v1;corev1";
+
+// Represents a single Envoy Cluster
+message Cluster {
```

Comment:
Should I nest this deeper into a `clutch.core.envoy.v1` package? Or keep in the current package and rename to `EnvoyCluster`?

Reply:
I like the idea of nesting this deeper into `clutch.core.envoy.v1` - I do not have a strong opinion when it comes to this versus renaming to `EnvoyCluster` but I think that adding `Envoy` *somewhere* makes sense to make it clear what kind of cluster we area dealing with especially since this is going to be a part of a generic `Core` package.

//cc @snowp  in case he has any opinion here.

Reply:
Yeah I like the former more. Let me update the PR and we can go from there.